pyproject.toml

```diff
@@ -31,17 +31,18 @@ authors = [
 ]
 dependencies = [
     "coverage>=7.2.0", # Code coverage tool. Sadly baked into every Case.
-    "h5py>=3.9 ; python_version >= '3.11.0'", # Needed because our database files are H5 format
     "h5py>=3.0,<=3.9 ; python_version < '3.11.0'",
+    "h5py>=3.9 ; python_version >= '3.11.0'", # Needed because our database files are H5 format
     "htmltree>=0.7.6", # Our reports have HTML output
     "matplotlib>=3.5.3,<3.8.0", # Important plotting library
     "numpy>=1.21", # Important math library
     "ordered-set>=3.1.1", # A useful data structure
     "pluggy>=1.2.0", # Central tool behind the ARMI Plugin system
     "pyDOE>=0.3.8", # We import a Latin-hypercube algorithm to explore a phase space
     "pyevtk>=1.2.0", # Handles binary VTK visualization files
-    "ruamel.yaml.clib ; python_version >= '3.11.0'", # C-based core of ruamel below
+    "python-dateutil>=2.2", # To read a datetime string more easily.
```

Comment:
lol one use in a function not used in our codebase. I can see one downstream project that uses this

https://github.com/terrapower/armi/blob/c078cde00832c559586cbcc255356bac445a7a30/armi/utils/parsing.py#L97-L99

Might be worth removing a scarcely used / trivial function that brings in an entire dependency? Not blocking for this. But maybe a follow on

Reply:
Oh, hey, that's a good catch.

Let me test that.  

Yeah, I'd love to rip that noise out.

Reply:
That's a good catch.

This function is only used in one project anywhere, that I can see.  It is used in a terrapower internal project called "nuclerdata-uq".  So... that's our limiting factor right now.

At the very least, we could move this method and dependency over to their repo. So I can tear this out of ARMI.

Reply:
Either is good for me. We can punt to something later or make the change here and work with that project to update.